data_steward/cdr_cleaner/cleaning_rules/clean_ppi_numeric_fields_using_parameters.py

```diff
@@ -1,131 +1,262 @@
-import constants.cdr_cleaner.clean_cdr as cdr_consts
+"""
+Apply value ranges to ensure that values are reasonable and to minimize the likelihood
+of sensitive information (like phone numbers) within the free text fields.
 
-CLEAN_PPI_NUMERIC_FIELDS = """
-UPDATE
-  `{project}.{dataset}.observation` u1
-SET
-  u1.value_as_number = NULL,
-  u1.value_as_concept_id = 2000000010
-FROM
-  (
-  SELECT
-  *
-FROM
-  `{project}.{dataset}.observation`
-WHERE
-  observation_concept_id = 1585889 AND (value_as_number < 0 OR value_as_number > 20)
+Original Issues: DC-827, DC-502, DC-487
 
-UNION ALL
+The intent is to ensure that numeric free-text fields that are not manipulated by de-id
+have value range restrictions applied to the value_as_number field across the entire dataset.
+"""
 
-SELECT
-  *
-FROM
-  `{project}.{dataset}.observation`
-WHERE
-  observation_concept_id = 1585890 AND (value_as_number < 0 OR value_as_number > 20)
+# Python imports
+import logging
 
-UNION ALL
+# Third party imports
+from jinja2 import Environment
 
-SELECT
-  *
-FROM
-  `{project}.{dataset}.observation`
-WHERE
-  observation_concept_id = 1585795 AND (value_as_number < 0 OR value_as_number > 99)
+# Project imports
+import constants.cdr_cleaner.clean_cdr as cdr_consts
+from cdr_cleaner.cleaning_rules.base_cleaning_rule import BaseCleaningRule
+from constants.bq_utils import WRITE_TRUNCATE
 
-UNION ALL
+LOGGER = logging.getLogger(__name__)
 
-SELECT
-  *
-FROM
-  `{project}.{dataset}.observation`
-WHERE
-  observation_concept_id = 1585802 AND (value_as_number < 0 OR value_as_number > 99)
+jinja_env = Environment(
+    # help protect against cross-site scripting vulnerabilities
+    autoescape=True,
+    # block tags on their own lines
+    # will not cause extra white space
+    trim_blocks=True,
+    lstrip_blocks=True,
+    # syntax highlighting should be better
+    # with these comment delimiters
+    comment_start_string='--',
+    comment_end_string=' --')
 
-UNION ALL
+SAVE_TABLE_NAME = 'dc_827_obs_changed_rows_saved'
```

Comment:
This can be generated programmatically from get_sandbox_table_names() helper function.

  ```
  def get_sandbox_tablenames(self):
        table_names = list()
        table_names.append(self._issue_numbers[0] + self._affected_tables[0])
        return table_names

```

Reply:
Modified `get_sandbox_tablenames` function
Changed to `intermediary_table=self.get_sandbox_tablenames()`
Removed `SAVE_TABLE_NAME = 'dc_827_obs_changed_rows_saved`